core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -718,6 +718,37 @@ class QualificationAppInfo(
       appId, hasFailures)
   }
 
+  def getAllSQLDurations: Seq[Long] = {
+    sqlIdToInfo.map { case (_, info) =>
+      info.rootExecutionID match {
+        // We return the duration if sqlId doesn't have a rootExecutionID or if the rootExecutionID
+        // is the same as the sqlId. In some cases, the child completes the execution before the
+        // parent, so we need to check if the child's duration is within the parent's duration.
+        // We add the child's duration to the parent's duration if it's not within the parent's
+        // duration.
+        case Some(rootExecutionID) if rootExecutionID == info.sqlID =>
+          info.duration.getOrElse(0L)
+        case Some(rootExecutionID) if rootExecutionID != info.sqlID =>
+          val rootExecutionInfo = sqlIdToInfo.get(rootExecutionID)
+          if (rootExecutionInfo.nonEmpty) {
+            val rootExecutionStartTime = rootExecutionInfo.get.startTime
+            val rootExecutionEndTime = rootExecutionInfo.get.endTime.getOrElse(0L)
+            val sqlStartTime = info.startTime
+            val sqlEndTime = info.endTime.getOrElse(0L)
+            if (sqlStartTime < rootExecutionStartTime || sqlEndTime > rootExecutionEndTime) {
+              info.duration.getOrElse(0L)
```

Comment:
We can add a comment here saying why this if condition is enough to check for overlap. Later we may forget why we are not applying stricter checks.
For example:
- the above check condition will be true if the child is not completely inside the root. nevertheless we still account for its total duration and not the overlap.
- the condition will be true if the child starts after the root ends. is this even a possible case?
- the condition will be true if the spans the root. is this possible case?